### ISSUE TITLE
Is it possible to use this to implement an mpris client?

### ISSUE THREAD

**Issue by nukeop** [sidorares/dbus-native#240]
Hi, I'm trying to move away from libdbus in my node program, and I'm wondering if it's possible to use this library as an mpris client? I only want to receive and send some basic messages like title, thumbnail, artist, and receive events like play/pause, next, and quit.

**Comment by sidorares** [comment]
should be possible (anything is possible at messaging level, just a bit more verbose to program)

**Comment by sidorares** [comment]
People definitely attempted to interact with mpris using this library, have a look at #102 #89 #121 #60

**Comment by sidorares** [comment]
Yes. MPRIS is an ordinary D-Bus interface, so everything you listed works with this library: reading `Metadata` (title/artist/art URL) via `org.freedesktop.DBus.Properties`, calling `Play`/`Pause`/`Next` on `org.mpris.MediaPlayer2.Player`, and subscribing to `PropertiesChanged` to get updates.

Sketch for the client side:

```js
const dbus = require('dbus-native');
const bus = dbus.sessionBus();

bus.getService('org.mpris.MediaPlayer2.vlc').getInterface(
  '/org/mpris/MediaPlayer2',
  'org.mpris.MediaPlayer2.Player',
  (err, player) => {
    if (err) throw err;
    player.Metadata(console.log);   // properties are callable with a callback
    player.PlayPause();
  }
);
```

Use `bus.listNames()` and filter for names starting with `org.mpris.MediaPlayer2.` to discover players.

Note that `Metadata` is `a{sv}`, so values come back as `[parsedSignature, [value]]` pairs rather than a plain object — see #3 / #132, and [ROADMAP §4.1](https://github.com/sidorares/dbus-native/blob/master/ROADMAP.md#41-variant-handling) for the plan to make that ergonomic.

If you want a worked example rather than a sketch, [`mpris-service`](https://github.com/emersion/mpris-service) implements the server side on top of a fork of this library.
